### PR TITLE
feat(admin): Institution showcase page

### PR DESCRIPTION
**Pull request by pybuche**
# First Iteration of the Institution Showcase page

- File uploader
- Institution URL
- Description text block

Screenshot 2022-07-05 at 14 54 13

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,9 @@ lerna-debug.log*
 # Diagnostic reports (https://nodejs.org/api/report.html)
 report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json
 
+# MacOS
```
Comment (ivangabriele) [inline .gitignore:16]: It's a better practice to use [global gitignore](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c) to ignore user-specific stuff 😉. It's like IDE settings, we eventually use the contributing document to suggest configs.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
 - [Node.js](https://nodejs.org)
 - [Yarn](https://yarnpkg.com/getting-started/install)
 - [Docker](https://www.docker.com/get-started)
+- `brew install gsed`
```
Comment (ivangabriele) [inline CONTRIBUTING.md:23]:
```suggestion
- `brew install gsed` # only for MacOS
```

File: app/molecules/AdminForm/Editor.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ export function Editor({ helper, isDisabled = false, label, name, placeholder }:
 
   const hasError = (touched[name] !== undefined || submitCount > 0) && Boolean(errors[name])
   const maybeError = hasError ? String(errors[name]) : undefined
-  const defaultVaLue = useMemo(() => values[name], [])
+  const defaultValue = values[name] || '<br>'
```
Comment (ivangabriele) [inline app/molecules/AdminForm/Editor.tsx:18]: In fact we should be able to pass an undefined there. I may deserve a TODO for later on.

https://github.com/singularity-ui/core/blob/main/fields/MarkdownEditor/helpers.ts#L19

File: app/molecules/AdminForm/FileUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  const uploadUrl = values?.[name]?.url
+
+  const handleFileChange = async event => {
+    setIsUploading(true)
+    try {
+      const file = event.target.files[0]
+      const { url } = await uploadToS3(file)
+
+      // TODO: Remove useless FileType
+      setFieldValue(name, { title: file.name, type: FileType.EXTERNAL, url })
```
Comment (ivangabriele) [inline app/molecules/AdminForm/FileUpload.tsx:68]: For the upload itself in the next PR, I suggest [formidable](https://www.npmjs.com/package/formidable) but you surely already know it. Indeed you won't need to store the file type with that: https://github.com/betagouv/tell-me/blob/main/pages/api/assets/index.ts#L31

**Review by ivangabriele — APPROVED** [review]
Approved because there is nothing blocking. I just made a few minor comments and suggestions.

File: app/molecules/AdminForm/FileUpload.tsx (hunk 4 above)
Comment (pybuche) [reply app/molecules/AdminForm/FileUpload.tsx:68]: We actually already have the type, however it doesn't fit the enum you've implemented... And I'm not really sure what's the purpose of keeping the extension stored in the DB, if we have the full URL. WDYT?

**Pull request merged**